src/frontend/Canonicalize.ml

```diff
@@ -1,95 +1,20 @@
-    ; ("binomial_coefficient_log", "lchoose")
-    ; ("integrate_ode", "integrate_ode_rk45") ]
-
-let deprecated_distributions =
-  String.Map.of_alist_exn
-    (List.concat_map Middle.Stan_math_signatures.distributions
-       ~f:(fun (fnkinds, name, _) ->
-         List.filter_map fnkinds ~f:(function
-           | Lpdf -> Some (name ^ "_log", name ^ "_lpdf")
-           | Lpmf -> Some (name ^ "_log", name ^ "_lpmf")
-           | Cdf -> Some (name ^ "_cdf_log", name ^ "_lcdf")
-           | Ccdf -> Some (name ^ "_ccdf_log", name ^ "_lccdf")
-           | Rng | UnaryVectorized -> None ) ))
-
-let deprecated_userdefined = String.Table.create ()
-
-let is_distribution name =
-  Option.is_some (String.Map.find deprecated_distributions name)
-
-let rename_distribution name =
-  Option.value ~default:name (String.Map.find deprecated_distributions name)
-
-let rename_function name =
-  Option.value ~default:name (String.Map.find deprecated_functions name)
-
-let distribution_suffix name =
-  String.is_suffix ~suffix:"_lpdf" name
-  || String.is_suffix ~suffix:"_lpmf" name
-  || String.is_suffix ~suffix:"_lcdf" name
-  || String.is_suffix ~suffix:"_lccdf" name
-
-let userdef_distributions stmts =
-  List.filter_map
-    ~f:(function
-      | {stmt= FunDef {funname= {name; _}; _}; _} ->
-          if
-            String.is_suffix ~suffix:"_log_lpdf" name
-            || String.is_suffix ~suffix:"_log_lpmf" name
-          then Some (String.drop_suffix name 5)
-          else if String.is_suffix ~suffix:"_log_log" name then
-            Some (String.drop_suffix name 4)
-          else None
-      | _ -> None)
-    (Option.value ~default:[] stmts)
-
-let without_suffix user_dists name =
-  if
-    String.is_suffix ~suffix:"_lpdf" name
-    || String.is_suffix ~suffix:"_lpmf" name
-  then String.drop_suffix name 5
-  else if
-    String.is_suffix ~suffix:"_log" name
-    && not
-         ( is_distribution (name ^ "_log")
-         || List.exists ~f:(( = ) name) user_dists )
-  then String.drop_suffix name 4
-  else name
-
-let rec repair_syntax_expr {expr; emeta} =
-  let expr =
-    match expr with
-    | FunApp (f, {name; id_loc}, e) when distribution_suffix name ->
-        CondDistApp (f, {name; id_loc}, List.map ~f:repair_syntax_expr e)
-    | CondDistApp (f, {name; id_loc}, e) when not (distribution_suffix name) ->
-        FunApp (f, {name; id_loc}, List.map ~f:repair_syntax_expr e)
-    | _ -> map_expression repair_syntax_expr ident expr
-  in
-  {expr; emeta}
-
-let repair_syntax_lval = map_lval_with repair_syntax_expr ident
+open Deprecation_analysis
 
 let rec repair_syntax_stmt user_dists {stmt; smeta} =
   match stmt with
   | Tilde {arg; distribution= {name; id_loc}; args; truncation} ->
       { stmt=
           Tilde
-            { arg= repair_syntax_expr arg
+            { arg
             ; distribution= {name= without_suffix user_dists name; id_loc}
-            ; args= List.map ~f:repair_syntax_expr args
```

Comment:
What happened to these lines btw?

Reply:
@nhuurre mentioned in the [original thread](https://github.com/stan-dev/stanc3/issues/600#issuecomment-692846128) that `repair_syntax_expr` is obsolete.